### DISCUSSION TITLE
connection.send should accept different data types as per spec

### DISCUSSION THREAD

**Discussion by majek** [sockjs/sockjs-node#291] [Ideas]
Spec says that `send` method should accept `string`, `Blob`, and `ArrayBuffer` data types as a parameter. We should support that.

**Comment by dvv** [comment]
+1

**Comment by majek** [comment]
Apparently, types `Blob` and `ArrayBuffer` are quite new. The support for this stuff was added to chrome like a month ago. We should indeed support this stuff, but it can be done later on. For now we should only support unicode.
- https://plus.google.com/115535723976198353696/posts/ERN6zYozENV
- https://developer.mozilla.org/en/JavaScript_typed_arrays
- https://developer.mozilla.org/en/DOM/Blob

**Comment by kapouer** [comment]
Isn't it easier to support those data types than unicode ? At least they are not hit by surrogates conversion problems.

**Comment by majek** [comment]
@kapouer: I don't know. We should support at least one of those two: proper unicode _or_ binary data. In the perfect world we would support both. I'm aware of that.

**Comment by Rob-pw** [comment]
Does SockJS now support: binary / blob / ArrayBuffer?

**Comment by pretannikai** [comment]
+1

**Comment by imseon** [comment]
Any plans to get this done?
